aiohttp_client_cache/backends/sqlite.py

```diff
@@ -57,45 +55,36 @@ def __init__(
     ):
         super().__init__(**kwargs)
         self.connection_kwargs = get_valid_kwargs(sqlite_template, kwargs)
+        self.connection_kwargs['check_same_thread'] = False
         self.filename = _get_cache_filename(filename, use_temp)
         self.table_name = table_name
 
         self._bulk_commit = False
-        self._initialized = False
-        self._connection = None
+        self._connection: Optional[aiosqlite.Connection] = None
         self._lock = RLock()
 
     @asynccontextmanager
-    async def get_connection(self, autocommit: bool = False) -> AsyncIterator[aiosqlite.Connection]:
-        db = (
-            self._connection
-            if self._connection
-            else await aiosqlite.connect(self.filename, **self.connection_kwargs)
-        )
-        try:
-            yield await self._init_db(db)
-            if autocommit and not self._bulk_commit:
-                await db.commit()
-        finally:
-            if not self._bulk_commit:
-                await db.close()
+    async def get_connection(self, commit: bool = False) -> AsyncIterator[aiosqlite.Connection]:
+        if not self._connection:
+            self._connection = await aiosqlite.connect(self.filename, **self.connection_kwargs)
```

Comment:
`connect` is asynchronous so you might end up calling it repeatedly and the context manager can yield before the database is initialised since it's only checking if the connection has been set.   I think a better approach would be:

```python
        if not self._connection:
            async with self._alock:
                if not self._connection:
                    connection = await aiosqlite.connect(self.filename, **self.connection_kwargs)
                    await self._init_db(connection)
                    self._connection = connection
```

(Apologies if you are aware, just thought I'd comment while the PR's still in draft cuz I might not get another chance.)

Reply:
Thanks, that's a good point. If you'd like, can ping you for review before merging, probably sometime later this week.

Reply:
Sure, thanks!

Reply:
I made a little progress on this, but wasn't able to finish it up this week. but I still plan to get to this soon.